src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -561,10 +558,14 @@ await _solutionLoadedEvent.WaitAsync()
             SolutionRestoreRequest request,
             CancellationToken token)
         {
+            // if the request is implicit & this is the first restore, assume we are restoring due to a solution load.
+            var isSolutionLoadRestore = Interlocked.CompareExchange(ref _solutionLoad, 1, 0) == 0 &&
```

Comment:
I think the `Interlocked.CompareExchange` is unnecessary. It's not a big deal, it's only one method call. But it requires more cognative load to understand than 
```cs
var isSolutionLoadRestore = _firstRestore && request.RestoreSource == RestoreOperatingSource.Implicit;
_firstRestore = false;
```

This method, `ProcessRestoreRequestAsync`, can never run concurrently, right? That's the point of this class? If that's right, we can treat it as if it's single threaded, and hence simple read and set operations work just as well as `Interlocked.CompareExchange`.

Reply:
Good point.